piptools/_compat/pip_compat.py

```diff
@@ -24,7 +28,7 @@ def do_import(module_path, subimport=None, old_path=None):
 PackageFinder = do_import('index', 'PackageFinder')
 FormatControl = do_import('index', 'FormatControl')
 Wheel = do_import('wheel', 'Wheel')
-Command = do_import('basecommand', 'Command')
-cmdoptions = do_import('cmdoptions')
+Command = do_import('cli.base_command', 'Command', old_path='basecommand')
```

Comment:
I think this doesn't work with Pip 18.0.

Reply:
It does.

Reply:
Using the new import shims it tries each possibility and falls back to the next ones

```python
>>> def do_import(module_path, subimport=None, old_path=None):
  2         old_path = old_path or module_path
  3         prefixes = ["pip._internal", "pip"]
  4         paths = [module_path, old_path]
  5         search_order = ["{0}.{1}".format(p, pth) for p in prefixes for pth in paths if pth is not None]
  6         package = subimport if subimport else None
  7         for to_import in search_order:
  8             if not subimport:
  9                 to_import, _, package = to_import.rpartition(".")
 10             print('Import target: %s\tpackage: %s' % (to_import, package))
>>> do_import('cli.base_command', 'Command', old_path='basecommand')
Import target: pip._internal.cli.base_command   package: Command
Import target: pip._internal.basecommand        package: Command
Import target: pip.cli.base_command     package: Command
Import target: pip.basecommand  package: Command
```

Reply:
And in case you need proof:

```console
 ~/g/pip-tools   pip18-support    vf tmp --python=python3.6
Running virtualenv with interpreter /home/hawk/.pyenv/shims/python3.6
Using base prefix '/home/hawk/.pyenv/versions/3.6.5'
New python executable in /home/hawk/.virtualenvs/tempenv-3369325970666/bin/python3.6
Also creating executable in /home/hawk/.virtualenvs/tempenv-3369325970666/bin/python
Installing setuptools, pip, wheel...done.
  ³ tempenv-3369325970666  ~/g/pip-tools   pip18-support    pip install -e .
Obtaining file:///home/hawk/git/pip-tools
[...]
Installing collected packages: click, first, six, pip-tools
  Running setup.py develop for pip-tools
Successfully installed click-6.7 first-2.0.1 pip-tools six-1.11.0
  ³ tempenv-3369325970666  ~/g/pip-tools   pip18-support    pip install ptpython
Collecting ptpython
  Downloading https://files.pythonhosted.org/packages/d0/dd/163a698a86b9de92857f128117034bdb36f5a784d839eea3bc07e2c858ae/ptpython-0.41-py3-none-any.whl (47kB)
    100% |████████████████████████████████| 51kB 1.1MB/s
[...]

  ³ tempenv-3369325970666  ~/g/pip-tools   pip18-support    ptpython
>>> from pip import __version__ as pip_version
>>> print(pip_version)
18.0
>>> from piptools._compat import Command
>>> Command
<class 'pip._internal.basecommand.Command'>
>>> 
```

Reply:
FYI if you are using the code for importing from the `pip_compat` file I recommend grabbing the updated version, it borrows some concepts from https://github.com/brettcannon/modutil to do imports a bit more efficiently

Reply:
Ah, OK, great that it works. Sorry for the incorrect analysis.

Good work! :+1: